lib/default.nix

```diff
@@ -263,37 +263,56 @@ let
       );
 
       checks = eachSystem (
-        { system, ... }:
-        lib.mergeAttrsList [
-          # add all the supported packages, and their passthru.tests to checks
-          (withPrefix "pkgs-" (
-            lib.concatMapAttrs (
-              pname: package:
-              {
-                ${pname} = package;
+        { system, pkgs, ... }:
+        lib.mergeAttrsList (
+          [
+            # add all the supported packages, and their passthru.tests to checks
+            (withPrefix "pkgs-" (
+              lib.concatMapAttrs (
+                pname: package:
+                {
+                  ${pname} = package;
+                }
+                # also add the passthru.tests to the checks
+                // (lib.mapAttrs' (tname: test: {
+                  name = "${pname}-${tname}";
+                  value = test;
+                }) (filterPlatforms system (package.passthru.tests or { })))
+              ) (filterPlatforms system (inputs.self.packages.${system} or { }))
+            ))
+            # build all the devshells
+            (withPrefix "devshell-" (inputs.self.devShells.${system} or { }))
+            # add nixos system closures to checks
+            (withPrefix "nixos-" (
+              lib.mapAttrs (_: x: x.config.system.build.toplevel) (
+                lib.filterAttrs (_: x: x.pkgs.system == system) (inputs.self.nixosConfigurations or { })
+              )
+            ))
+            # add darwin system closures to checks
+            (withPrefix "darwin-" (
+              lib.mapAttrs (_: x: x.system) (
+                lib.filterAttrs (_: x: x.pkgs.system == system) (inputs.self.darwinConfigurations or { })
+              )
+            ))
+          ]
+          ++ (lib.optional (inputs.self.lib.tests or { } != { }) {
+            lib-tests = pkgs.runCommandLocal "lib-tests" { nativeBuildInputs = [ pkgs.nix-unit ]; } ''
+              export HOME="$(realpath .)"
```

Comment:
Just curious as why this and the NIX_CONFIG below are needed/useful here? Is it to keep nix-unit caches local to the current project? 

And couldn't that lead to unexpected behavior insofar as it's different from just running "nix-unit" in your shell locally?

Reply:
What I was trying to solve here is that because the build runs in a sandbox, it cannot access the user's home or talk to the nix-daemon. So Instead I'm setting a fake home (because nix-unit stores things in the user's home), and bring all the flake inputs manually (so they become drv references, and available to nix)

Reply:
Ah of course, not sure how i missed the sandboxed nature of a runLocalCommand inside a sandboxed check here earlier.